pkg/engine/virtualmachinescalesets.go

```diff
@@ -85,7 +85,7 @@ func CreateMasterVMSS(cs *api.ContainerService) VirtualMachineScaleSetARM {
 	addCustomTagsToVMScaleSets(cs.Properties.MasterProfile.CustomVMTags, &virtualMachine)
 
 	if hasAvailabilityZones {
-		virtualMachine.Zones = &masterProfile.AvailabilityZones
+		virtualMachine.Zones = to.StringPtr("[parameters('availabilityZones')]")
```

Comment:
the `Zones` property is a `*[]string` (pointer to a string slice), so this would be the syntax for expressing the above change as a literal value:

`&[]string{"[parameters('availabilityZones')]"}`

Reply:
fixed

Reply:
Do you think `availabilityZones` should be `AvailabilityZones`?

Reply:
do you think this can be the fix? ` &[]string{"[string(parameters('availabilityZones')]"}`

Reply:
do you think this can be the fix? &[]string{"[string(parameters('availabilityZones'))]"}

Reply:
do you think this can be the fix? `&[]string{"[string(parameters('availabilityZones'))]"}`